### PR TITLE
feat/SingleRequest

### PR DESCRIPTION
**Pull request by FernandoGrangeiro**
Colocado o BaseService para ter requisições que voltam um Single para o caso de precisar do que é retornado sem que esteja no realm

### CODE REVIEW

File: os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt [hunk 1]
```diff
@@ -16,17 +16,16 @@ import retrofit2.HttpException
  */
 
 open class BaseService {
+
     companion object {
 
-        fun <ApiModel: BaseApiModel<AppModel>, AppModel> makeApiCall(
+        fun <ApiModel : BaseApiModel<AppModel>, AppModel> makeApiCallCompletable(
```
Comment (rudieros) [inline os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt]: I don't really like those names for the method (makeApiCallCompletable and makeApiCallSingle), can't think of a better one right now, but we should work on it!

File: os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt [hunk 2]
```diff
@@ -46,21 +44,64 @@ open class BaseService {
...
+                remote: Single<OSResponse<ApiModel>>,
+                local: ((appModel: AppModel) -> Completable)? = null): Single<AppModel> {
+            return Single.create {
+                val d = DisposableManager()
+                it.setDisposable(d)
+                remote.subscribeOn(Schedulers.io())
+                        .observeOn(Schedulers.io())
+                        .subscribe({ response ->
+                            when {
+                                response.error != null -> it.onError(response.error)
```
Comment (rudieros) [inline os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt]: isn't this whole behavior being repeated? wouldn't be better to have a method that does this?

File: os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt [hunk 3]
```diff
@@ -46,21 +44,64 @@ open class BaseService {
...
+                                        local?.invoke(data)?.subscribeOn(OSApplication.realmScheduler)?.blockingAwait()
+                                    } catch (throwable: Throwable) {
+                                        it.onError(throwable)
                                     }
-                                }).collect(d)
+                                    it.onSuccess(response.data.mapToAppModel())
+                                }
+                            }
+                        }, { t ->
+                            if (t is HttpException) {
```
Comment (rudieros) [inline os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt]: same for this part

File: os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt (hunk 1 above)
Comment (MatheusFV) [reply os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt]: Agree

**Review by MatheusFV — APPROVED** [review]

File: os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt (hunk 3 above)
Comment (FernandoGrangeiro) [reply os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt]: Ok

File: os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt (hunk 2 above)
Comment (FernandoGrangeiro) [reply os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt]: Ok

File: os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt (hunk 1 above)
Comment (FernandoGrangeiro) [reply os-arturito/src/main/java/com/outsmart/os_arturito/BaseService.kt]: I'll update this on next PR

**Review by rudieros — APPROVED** [review]
looks good!

**Pull request merged**